app/static/admin/js/pages/gen_form/form-donor.js

```diff
@@ -1,139 +1,165 @@
+        this.input.telephone.value = data.telephone_number;
+        this.input.mobile.value = data.mobile_number;
+        this.input.ref.value = data.customer_ref;
+        this.input.needReceipt.value = data.want_receipt;
+        this.input.address.value = data.address_line;
+        this.input.city.value = data.city;
+        this.input.province.value = data.province;
+        this.input.postalCode.value = data.postal_code;
+
+        util.setButton(this.button, "existing");
+
+        donation.getDonation(data.id);
+    }.bind(dom);
 
     var store = {};
+
     /**
      * REQUIRE: this.donor.input.name == name field
      * EFFECT: calls for a list of names
      */
     function getNames(request, response) {
-      $.ajax({
-        url: "/api/autocomplete_name",
-        dataType: "json",
-        data: {
-          key: this.dom.input.name.value
-        },
-        success: function(data) {
-            store = {};
-            data.reduce(function(store, donor) {
+        $.ajax({
+            url: "/api/autocomplete_name",
+            dataType: "json",
+            data: {
+                key: dom.input.name.value
+            },
+            success: function (data) {
+                store = {};
+                data.reduce(function (store, donor) {
+                    var str = donor.donor_name + ', ' + donor.id;
+                    store[str] = donor;
+                    return store;
+                }, store);
+                response(Object.keys(store));
+            },
+            error: function () {
+                console.error(arguments);
+            }
+        });
+    }
+
+    /**
+     *
+     */
+    function updateDonor() {
+
+        var serializedData = $(dom.form).serialize();
+        serializedData.donor_name = serializedData.donor_name.split(",")[0];
+
+        $.ajax({
+            beforeSend: util.csrf,
+            url: "/api/donor",
+            type: "PUT",
+            dataType: "json",
+            data: $(dom.form).serialize(),
+            success: function (donor) {
+                alert('success! donor id: ' +  donor.id);
+                store = {};
                 var str = donor.donor_name + ', ' + donor.id;
                 store[str] = donor;
-                return store;
-            }, store);
-          response(Object.keys(store));
-        },
-        error: function() {
-          console.error(arguments);
-        }
-      });
+                dom.input.name.value(str);
+            },
+            error: function () {
+                console.error(arguments);
+            }
+        });
+    }
+
+    /**
+     * delete donor using id
+     */
+    function deleteDonor() {
+        $.ajax({
+            beforeSend: util.csrf,
+            url: "/api/donor",
+            type: "DELETE",
+            dataType: "json",
+            data: { donor_id: form.input.id.value },
+            success: function (donor) {
+                alert('success!');
```

Comment:
A better alert message would be nice. Something like "Donor [donor_name] has been successfully deleted."

Reply:
It was a temporary thing for a test purpose, but sure.